core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -89,6 +90,8 @@ protected QueryShardContext initialValue() {
     private final ParseFieldMatcher parseFieldMatcher;
     private final boolean defaultAllowUnmappedFields;
 
+    private TermsLookupFetchService termsLookupFetchService;
```

Comment:
Geo and MLT all use a fetch service, so this is being consistent with these queries.